### PR TITLE
Modernised script for Python 3 etc

### PR DESCRIPTION
**Pull request by SandyThomson**
## TL;DR

This PR modifies this script to make it much less of a faff to install and use on a modern linux distro

## Waffle

I've been using the existing `python2` script (in `master`) for around a year using Debian 10. It was a faff to get working from memory. I decided to reflash the OS to the latest Debian stable version (which uses `python3` by default).

I anticipated it was going to be a world of pain to get this script working again using `python2` on Debian, due to updated Bluetooth stack, missing python 2 libraries etc.

I decided to look into what it would take to upgrade the script to Python 3 .. even though I've not used Python much.

Pleasantly it wasn't too difficult, it took an evening working out what had changed etc, and it now pretty much works out the box, which is much better than having to handcraft very old libraries onto your system and hoping for the best.

Thanks again for producing this script. The WiiBoard is great, I use it and this software daily.

### CODE REVIEW

**Comment by skorokithakis** [comment]
This is great, thanks! However, why did you delete the wiibind script? Also, how do you get around the problem of having to turn the board around every time to press the red button?

**Comment by SandyThomson** [comment]
I believe the linux kernel now supports these devices natively, so no need for the wiibind script any more. Also, due to the updated bluetooth stack, the disconnection by python seems to be cleaner now (I didn't need it when I used it with `python2`, mind you).

See how you get on with it.

**Comment by skorokithakis** [comment]
Ah, thanks. Did you get around the "pairing every time" issue?

**Comment by SandyThomson** [comment]
I think the new Bluetooth stack supports the way that the Wii Balance board pairs (I forget the technical term for what it's called now ...).
Like, it just works, you don't need any manual bluetooth pairing steps _at all_.

**Comment by skorokithakis** [comment]
Oh wow, alright, I'm going to try this immediately, thanks!

**Comment by skorokithakis** [comment]
Hmm, did this work for you? I'm always getting this:

```
Requesting calibration
Traceback (most recent call last):
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 297, in <module>
    main()
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 286, in main
    board.connect(address)  # The wii board must be in sync mode at this time
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 109, in connect
    self.calibrate()
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 262, in calibrate
    self.send(message)
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 245, in send
    self.controlsocket.send(bytes.fromhex(updatedHex))
  File "<string>", line 3, in send
SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats
```

**Comment by SandyThomson** [comment]
What version of python are you running?
```
$ python --version
Python 3.9.2
```

**Comment by SandyThomson** [comment]
Maybe I should be explicitly setting `python3` at the top of the script?

**Comment by skorokithakis** [comment]
Ah, apparently the Ubuntu version of PyBluez needs updating. It worked after I updated, though it still refuses to connect without the red button:

```
01:59:39 $ ./gr8w8upd8m8.py 00:1E:35:C2:5E:F0
Trying to connect...
Traceback (most recent call last):
  File "<string>", line 3, in connect
_bluetooth.error: (112, 'Host is down')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 297, in <module>
    main()
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 286, in main
    board.connect(address)  # The wii board must be in sync mode at this time
  File "/home/stavros/Desktop/gr8w8upd8m8/gr8w8upd8m8.py", line 103, in connect
    self.receivesocket.connect((address, 0x13))
  File "<string>", line 5, in connect
bluetooth.btcommon.BluetoothError: [Errno 112] Host is down
```

Does that work for you?

**Comment by skorokithakis** [comment]
(Yes, it did need `python3` at the top, but that was an easy fix)

**Comment by SandyThomson** [comment]
I never specify the address, so that's a use case I've not tested. For me, discovery is near instantaneous, although it might be a problem if you had >1 board on and synching at once.

**Comment by skorokithakis** [comment]
What does a typical run without the button look like for you? For me it insists I need to press it.

**Comment by SandyThomson** [comment]
So wait, there is a way to use this without having to press the red button??

**Comment by skorokithakis** [comment]
Aw man, so you *are* pressing the button every time? :stuck_out_tongue:

The first version didn't need it, you needed to press it to pair once, and afterwards it would work with just specifying the address (with the front button, like a Wii). A Bluez update broke that, and I've been trying to get it back ever since, as the board is not very useful without that.

**Comment by SandyThomson** [comment]
I'll have to park this until tomorrow (today now). I've got an early start. 

Does the address specification on the command line mean you don't need to press the sync button? If so, that's worthwhile as it would save me a few seconds every day (plus I could put the battery flap back on)

**Comment by skorokithakis** [comment]
Yes, exactly (or, well, it's supposed to).

We're lucky, I just got a Bluetooth Xbox controller today and saw some `bluetoothctl` commands on how to pair it. I tried them now and they worked with the board too with the front button, so there is some hope yet!

**Comment by skorokithakis** [comment]
But yes, I'll have to park this until tomorrow as well. Thanks!

**Comment by SandyThomson** [comment]
I'm convinced there will be a way to make it pair if necessary. Even if it has to shell out to bluetoothctl or whatever. OK, tomorrow.

**Comment by SandyThomson** [comment]
I've not forgotten about this, I've just been busy. Hopefully we'll get a chance to look at this this week.

**Comment by SandyThomson** [comment]
So between home automation projects and Crusader Kings 3, I've not spent any more time on this (yet). I need to wait for things to calm down a bit before looking at it again. 

The immediate problem for me has gone so therefore the motivation for spending time on this has somewhat gone. 

The red sync button press isn't an issue as I store it vertically and am lifting it all the time anyway.

**Comment by SandyThomson** [comment]
OK, I spent an hour on this tonight and didn't get very far with it. Even with using bluetoothctl, pairing and trusting the device, I can't get it to work without pressing the sync button. I had a look at the python libraries for bluetooth but they seem pretty basic. 

I'm happy with this for myself but I've clearly broken it for your usage. Not really sure how to proceed. Maybe we just abandon this PR?

**Comment by skorokithakis** [comment]
Well, the thing is that using it without the sync button has been broken for some time, and this PR modernizes it, at least, so I'd be ok to merge if you are.

**Comment by SandyThomson** [comment]
WFM 👍👍
